app/src/androidTest/java/ch/epfl/cs311/wanderwave/ui/BeaconScreenTest.kt

```diff
@@ -17,15 +24,28 @@ class BeaconScreenTest {
   @get:Rule val mockkRule = MockKRule(this)
 
   @RelaxedMockK private lateinit var mockNavigationActions: NavigationActions
-  private val viewModel = BeaconViewModel()
+  @RelaxedMockK private lateinit var beaconConnection: BeaconConnection
 
   @Before
   fun setup() {
+
+    val beaconFlow =
+        flowOf(
+            Beacon(
+                "UAn8OUadgrUOKYagf8a2",
+                Location(46.519653, 6.632273, "Lausanne"),
+                listOf<Track>(Track("Some Track ID", "Track Title", "Artist Name")),
+            ))
+    coEvery { beaconConnection.getItem(any<Beacon>()) } returns beaconFlow
+    coEvery { beaconConnection.getItem(any<String>()) } returns beaconFlow
```

Comment:
This is because coEvery is for coroutine stuff, and I put it when it has something to do with coroutines, which it absolutely does, dealing with task in the unit test is an anoying mess.
It's not the same, the function is overloaded and I just couldn't put any because it was overloaded so I had to put both.